src/adams_methods.jl

```diff
@@ -0,0 +1,636 @@
+
+                # Harier equation 7.4: If ||LE_{k+1}|| =< 1
+                # then step size is accepted and use next_dt for next step
+                # Else, then step size is rejected, reattempt with fac*next_dt
+                if err_kp1 <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    dt = .75*next_dt # make sure fac < 1
+                end
+            end # End of error check for current step size dt
+        end # End of loop for finding successfull step size for nth step
+            # repeats if success_step == false
+
+        #Keep step size constant if `dt` is close to `prev_dt`
+        if abs(next_dt/prev_dt)<2 && abs(next_dt/prev_dt)>=1
+            consec_const_steps += 1
+            next_dt = prev_dt
+        else
+            consec_const_steps = 0
+        end
+
+        #Don't step past `tfinal`
+        if tdir*(t_np1 + next_dt) >= tdir*(tfinal)
+            next_dt = tfinal-t_np1
+        end
+
+        #--#####################################################################
+        # Interpolation for dense output
+        #--#####################################################################
+        # interpolate specified points
+        while iter <= length(tspan) && tdir*tspan[iter]<tdir*t_np1
+            h_iter = tspan[iter]-t[n]
+
+            # Calculation of g_{j}(n)coefficients for j=0,...,k+1;
+            g_coefs!(g,c,dt,t,tspan[iter],n,min(n,steporder+1))
+
+            # Calculation of  ϕstar_{j}(n), and ϕ_{j}(n) coef. for j=0,...,k;
+            ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar,dy[n],t,tspan[iter],n,1:min(max(n-1,1),steporder+1))
+
+            yout_iter = copy(y[n])
+            for j= 1:steporder
+                for i= 1:dof
+                    yout_iter[i] += h_iter*g[j]*ϕstar[2,j][i]
+                end
+            end
+
+            #CE iteration for more precise interpolation
+            if n>=steporder+1
+                dy_iter=F(tspan[iter],yout_iter)
+                ϕ_coefs!(ϕ, ϕstar,dy_iter,1:steporder+1)
+                for i=1:dof
+                    yout_iter[i] += h_iter*g[steporder+1]*ϕ[3,steporder+1][i]
+                end
+            end
+
+            push!(tout,tspan[iter])
+            push!(yout, copy(yout_iter))
+            iter += 1
+        end
+
+        if iter <= length(tspan) && tspan[iter] == t_np1
+            # handling when tspan and choose time steps overlap
+            push!(tout, copy(t_np1))
+            push!(yout, copy(y_np1))
+            iter += 1
+        elseif points == :all
+            push!(tout, copy(t_np1))
+            push!(yout, copy(y_np1))
+        end
+
+        # restore values of coefficients for choosen step size
+        g_coefs!(g,c,dt,t,t_np1,n,min(n,steporder+1))
+        ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar,dy[n],t,t_np1,n,1:min(max(n-1,1),steporder+1))
+        ϕ_coefs!(ϕ, ϕstar,dy_np1,1:steporder+1)
+
+        # push work variables
```

Comment:
This has nothing to do with dense output, maybe move to above dense output loop?